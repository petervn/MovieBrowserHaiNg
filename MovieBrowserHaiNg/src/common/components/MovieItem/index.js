import React from 'react';
import {Image, View, Text, TouchableOpacity} from 'react-native';
import {getImage} from '../../../common/utils';
import CONSTANTS from '../../../constants';
import Styles from './style';

export default MovieItem = ({data: item = {}, horizontal, onPress}) => {
  const {
    release_date = '---',
    original_title = '',
    poster_path,
    character,
  } = item;
  return (
    <TouchableOpacity onPress={onPress} key={item.id} style={{width: 140}}>
      {!horizontal && (
        <View style={{margin: 10, marginBottom: 5}}>
          <View style={{flexDirection: 'row'}}>
            <Image style={Styles.image} source={getImage(poster_path)} />
            <View style={{flexDirection: 'column', marginLeft: 10}}>
              <Text numberOfLines={3} style={{fontSize: 17}}>
                {original_title}
              </Text>
              <View style={Styles.rowView}>
                <Text>{CONSTANTS.Strings.RELEASE_DATE}</Text>
                <Text>{release_date}</Text>
              </View>
            </View>
          </View>
          <View style={Styles.lineView} />
        </View>
      )}
      {horizontal && (
        <View style={{margin: 2, marginBottom: 5}}>
          <View style={{flexDirection: 'column', alignItems: 'center'}}>
            <Image style={Styles.image} source={getImage(poster_path)} />
            <View style={{flexDirection: 'column', alignItems: 'center'}}>
              <Text numberOfLines={1} style={{fontSize: 17}}>
                {original_title}
              </Text>
              <Text>{character}</Text>
              <Text>({release_date.split('-')[0]})</Text>
            </View>
          </View>
        </View>
      )}
    </TouchableOpacity>
  );
};
