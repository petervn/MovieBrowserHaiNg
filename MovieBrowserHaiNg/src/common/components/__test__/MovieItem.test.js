import React from 'react';
import {Text} from 'react-native';
import MovieItem from '../MovieItem';
import {shallow, mount} from 'enzyme';
import {expect as chaiExpect} from 'chai';

describe('MovieItem', () => {
  const data = {
      release_date: '2019-12-10',
      original_title: 'original_title',
      poster_path: 'poster_path',
      character: 'Marvel',
    },
    horizontal = true,
    onPress = () => {};

  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(
        <MovieItem data={data} horizontal={horizontal} onPress={onPress} />,
      );
      expect(component).toMatchSnapshot();
    });

    it('should render title', () => {
      let component = mount(
        <MovieItem data={data} horizontal={horizontal} onPress={onPress} />,
      );
      chaiExpect(component.find(Text)).to.have.lengthOf(3);
      const textValue = component
        .find(Text)
        .at(0)
        .text();
      expect(textValue).toEqual(data.original_title);
    });

    it('should render release date', () => {
      let component = mount(
        <MovieItem data={data} horizontal={horizontal} onPress={onPress} />,
      );
      const textValue = component
        .find(Text)
        .at(2)
        .text();
      expect(textValue).toEqual('(2019)');
    });
  });
});
