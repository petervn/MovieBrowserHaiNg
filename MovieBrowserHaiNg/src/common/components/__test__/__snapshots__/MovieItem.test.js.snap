// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieItem Rendering should match to snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "width": 140,
    }
  }
>
  <View
    style={
      Object {
        "margin": 2,
        "marginBottom": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://image.tmdb.org/t/p/w500poster_path",
          }
        }
        style={
          Object {
            "height": 180,
            "marginLeft": 5,
            "width": 120,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "fontSize": 17,
            }
          }
        >
          original_title
        </Text>
        <Text>
          Marvel
        </Text>
        <Text>
          (
          2019
          )
        </Text>
      </View>
    </View>
  </View>
</TouchableOpacity>
`;
